tests/integration/cluster_fixture.py

```diff
@@ -0,0 +1,226 @@
+
+    @classmethod
+    def _topic_conf(cls, options=None):
+        conf = {'num_partitions': 1,
+                'replication_factor': 1}
+
+        if options is not None:
+            conf.update(options)
+
+        return conf
+
+    def producer(self, options=None):
+        """
+        Returns a producer bound to this cluster.
+
+        :param dict options: additional producer configuration values to be used.
+        :returns: a new producer instance.
+        :rtype: confluent_kafka.Producer
+        """
+        return Producer(self.client_conf(options))
+
+    def consumer(self, options=None):
+        """
+        Returns a consumer bound to this cluster.
+
+        :param dict options: additional consumer values to be used.
+        :returns: a new consumer instance.
+        :rtype: confluent_kafka.Consumer
+        """
+        if options is None or 'group.id' not in options:
+            options.update({'group.id': uuid1()})
+
+        return Consumer(self.client_conf(options))
+
+    def create_topic(self, name, topic_options=None):
+        """
+        Creates a new topic with this cluster.
+
+        :param str name: topic name
+        :param dict topic_options: additions/overrides to topic configuration.
+        :returns: The topic's name
+        :rtype: str
+        """
+        if self._admin is None:
+            self._admin = AdminClient(self.client_conf())
+
+        future_topic = self._admin.create_topics([NewTopic(name, **self._topic_conf(topic_options))])
+
+        future_topic.get(name).result()
+        return name
+
+    def seed_topic(self, topic, value_source=None, key_source=None, header_source=None):
+        """
+        Populates a topic with data asynchronously.
+
+        If there are fewer keys or headers than values the index will be wrapped and reused with
+        each subsequent produce request.
+
+        :param str topic: topic to produce test data to.
+        :param list value_source: list or other iterable containing message key data to be produced.
+        :param list key_source: list or other iterable containing message value data to be produced.
+        :param list(dict) header_source: headers to attach to test data.
+        """
+
+        if self._producer is None:
+            self._producer = Producer(self.client_conf({'linger.ms': 500}))
+
+        if value_source is None:
+            value_source = ['test-data{}'.format(i) for i in range(0, 100)]
+
+        if key_source is None:
+            key_source = [None]
+
+        produce_thread = threading.Thread(target=_produce,
+                                          args=(self._producer, topic, value_source,
+                                                key_source, header_source))
+
+        produce_thread.daemon = True
+        produce_thread.start()
+
+    @property
+    def schema_registry(self):
+        return self._cluster.sr.get('url')
+
+
+def _produce(producer, topic, value_source, key_source, header_source=None):
```

Comment:
nit: Why the _source suffix? It is ambigious: what is a source? is it a generator?
`values, keys, headers` is clearer, simply indicating plural.

Reply:
also: doc strings